### PR TITLE
Hintfmt percent test, and fix

### PR DESCRIPTION
**Pull request by bburdette**
In light of the issue from https://github.com/NixOS/nix/pull/3720, I thought adding some tests would be a good idea.  

Also there's a tentative solution here... its maybe a little inelegant to search-and-replace on percent, but its simpler than some other things I thought of.  Still thinking though.

### CODE REVIEW

File: src/libutil/fmt.hh [hunk 1]
```diff
@@ -136,4 +139,11 @@ inline hintformat hintfmt(const std::string & fs, const Args & ... args)
     return f;
 }
 
+inline hintformat hintfmt(std::string fs)
+{
+    // we won't be receiving any args in this case, so escape all percents.
+    boost::replace_all(fs, "%", "%%");
+    hintformat f(fs);
+    return f;
+}
```
Comment (Ericson2314) [inline src/libutil/fmt.hh:148]: Nit, but I think I rather do
```suggestion
inline hintformat hintfmt(std::string plain_string)
{
    // we won't be receiving any args in this case, so just print the original string
    return hintformat { "%s", plain_string };
}
```
which has the slight advantage of not relying on the format string format, even though that format is unlikely to change.

File: src/libutil/fmt.hh [hunk 2]
```diff
@@ -136,4 +139,11 @@ inline hintformat hintfmt(const std::string & fs, const Args & ... args)
     return f;
 }
 
+inline hintformat hintfmt(std::string fs)
+{
+    // we won't be receiving any args in this case, so escape all percents.
+    boost::replace_all(fs, "%", "%%");
```
Comment (edolstra) [inline src/libutil/fmt.hh:145]: Hm, why is this needed? Can't we just do `return hintfmt("%s", fs)` here?

File: src/libutil/fmt.hh (hunk 1 above)
Comment (bburdette) [reply src/libutil/fmt.hh:148]: nice!  I like that better too.
